Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -240,44 +243,48 @@ bool PPCSymbolDB::LoadMap(const std::string& filename, bool bad)
 
     if (strcmp(temp, "UNUSED") == 0)
       continue;
-    if (strcmp(temp, ".text") == 0)
-    {
-      started = true;
-      continue;
-    };
-    if (strcmp(temp, ".init") == 0)
+
+    // Support CodeWarrior and Dolphin map
+    if (StringEndsWith(line, " section layout\n") || strcmp(temp, ".text") == 0 ||
+        strcmp(temp, ".init") == 0)
     {
-      started = true;
+      section_name = temp;
       continue;
-    };
+    }
+
+    // Skip four columns' header.
+    //
+    // Four columns example:
+    //
+    // .text section layout
+    //   Starting        Virtual
+    //   address  Size   address
+    //   -----------------------
     if (strcmp(temp, "Starting") == 0)
       continue;
-    if (strcmp(temp, "extab") == 0)
-      continue;
-    if (strcmp(temp, ".ctors") == 0)
-      break;  // uh?
-    if (strcmp(temp, ".dtors") == 0)
-      break;
-    if (strcmp(temp, ".rodata") == 0)
-      continue;
-    if (strcmp(temp, ".data") == 0)
-      continue;
-    if (strcmp(temp, ".sbss") == 0)
-      continue;
-    if (strcmp(temp, ".sdata") == 0)
-      continue;
-    if (strcmp(temp, ".sdata2") == 0)
-      continue;
     if (strcmp(temp, "address") == 0)
       continue;
     if (strcmp(temp, "-----------------------") == 0)
       continue;
-    if (strcmp(temp, ".sbss2") == 0)
-      break;
-    if (temp[1] == ']')
+
+    // Skip link map.
+    //
+    // Link map example:
+    //
+    // Link map of __start
+    //  1] __start(func, weak) found in os.a __start.c
+    //   2] __init_registers(func, local) found in os.a __start.c
+    //    3] _stack_addr found as linker generated symbol
+    // ...
+    //           10] EXILock(func, global) found in exi.a EXIBios.c
+    if (StringEndsWith(temp, "]"))
```

Comment:
"EndsWith", rather than "contains"?

Reply:
If I'm not mistaken the ```temp``` variable is the first word that doesn't contain any whitespace which is parsed with sscanf.
>**man (3) sscanf**
**%s**
Matches a sequence of non-white-space characters; the next pointer must be a pointer to character array that is long enough to hold the input sequence and the terminating null byte ('\0'), which is added automatically. The input string stops at white space or at the maximum field width, whichever occurs first.

Reply:
Ah, my bad, I had missed the sscanf call. LGTM then.